components/prism-glimmer.js

```diff
@@ -0,0 +1,290 @@
+			},
+			{
+				pattern: /^=/,
+				alias: 'attr-equals'
+			},
+			{
+				pattern: /\/?>/,
+			},
+		],
+	};
+
+	const FunctionName = {
+		'function-name': [
+			{
+				pattern: new RegExp(parenOpen + symbol),
+				lookbehind: true
+			},
+			{
+				pattern: new RegExp(mustacheOpen + symbol),
+				lookbehind: true
+			}
+		],
+	};
+
+	const Keywords = {
+		'builtin': toKeywords([_MODIFIERS, _SPECIAL, _DEBUG_HELPERS, _BLOCK_HELPERS].join(' ')),
+		'keyword': toKeywords([_INLINE_HELPERS, _LITERALS].join(' ')),
+		'operator': toKeywords([_EQUALITY_HELPERS, _NUMERIC_COMPARISON_HELPERS, _LOGICAL_OPERATOR_HELPERS, 'as'].join(' ')),
+	};
+
+	const Helper = {
+		'function': {
+			greedy: true,
+			pattern: /\([\S-_\d]+\b/,
+			inside: {
+				...Punctuation,
+				...FunctionName,
+				...Keywords,
+			}
+		}
+	};
+
+	const ThisExpression = {
+		'this-expression': {
+			pattern: /this\.[\S]+/,
+			lookbehind: true,
+			greedy: true,
+			inside: {
+				...Punctuation,
+				'namespace': /this/,
+				'property': /[\S]+/,
+			}
+		}
+	};
+
+	const MemberExpression = {
+		'member-expression': {
+			pattern: /[\S]+\.[\S]+/,
+			lookbehind: true,
+			greedy: true,
+			inside: {
+				...Punctuation,
+				'constant': /[\S]+/,
+				'property': /[\S]+/,
+			}
+		}
+	};
+
+	const INSIDE_MUSTACHE_AND_SUB_EXPRESSION = {
+		...Helper,
+		...Punctuation,
+		...ThisExpression,
+		...MemberExpression,
+		...Arg,
+		'number': NUMBER,
+		'boolean': /\b(?:true|false)\b/,
+		...Keywords,
+		...FunctionName,
+		'attr-name': /^[^=]+=/,
+		'string': STRING,
+		'variable': /\b[A-Za-z0-9_-]+\b/,
+	};
+
+	const SubExpression = {
+		'sub-expression': {
+			alias: 'punctuation',
+			pattern: /\([^)]+\)/,
+			lookbehind: true,
+			greedy: true,
+			inside: INSIDE_MUSTACHE_AND_SUB_EXPRESSION
+		}
+	};
+
+	const Mustache = {
+		'mustache': {
+			pattern: /\{\{\{?\/?[^}]+?\}?\}\}/,
+			lookbehind: true,
+			alias: 'punctuation',
+			greedy: true,
+			inside: {
+				...SubExpression,
+				...INSIDE_MUSTACHE_AND_SUB_EXPRESSION,
+			},
+		},
+	};
+
+	const String = {
+		'string': {
+			pattern: STRING,
+			inside: Mustache,
+		},
+	};
+
+	INSIDE_MUSTACHE_AND_SUB_EXPRESSION['string'] = String.string;
+
+	const {markup} = Prism.languages;
+
+	if (!markup) {
+		throw new Error('prism-markup is required');
+	}
+
+	Prism.languages.glimmer = {
+		'comment': [
+			{
+				pattern: /\{\{!--[\s\S]*?--\}\}/,
+			},
+			{
+				pattern: /\{\{![\s\S]*?\}\}/,
+			},
+		],
+		'number': NUMBER,
+		...Mustache,
+		tag: {
+			...markup.tag,
+			inside: {
+				'number': NUMBER,
+				...Arg,
+				...Mustache,
+				'tag': {
+					...markup.tag.inside.tag,
+					inside: {
+						...Punctuation,
+						'class-name': new RegExp(COMPONENT_NAME),
+					}
+				},
+				'attr-name': {
+					pattern: /\b[^=\b]+=/,
+					inside: {
+						...String,
+						...Punctuation,
+						...Arg,
+						...Mustache,
+					},
+				},
+				...Punctuation,
+				...String,
+			},
+		},
+	};
+
+
+	// if (Prism.languages.markup) {
```

Comment:
saving these for another PR as this one is pretty large

Reply:
Looking forward to it.

But please remove the commented-out code and make something like a TODO comment instead.